### PR TITLE
Add dfg as a pre-commit hook.

### PR DESCRIPTION
**Pull request by vyasr**
This change allows local and remote runs to handle calls to dfg identically, and removes the need for a separate CI check.

### CODE REVIEW

**Comment by vyasr** [comment]
Converting to draft pending some discussions on https://github.com/rapidsai/shared-action-workflows/issues/46.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -51,3 +51,7 @@ repos:
         language: python
         pass_filenames: false
         additional_dependencies: [gitpython]
+  - repo: https://github.com/rapidsai/dependency-file-generator
+    rev: v1.2.0
```
Comment (vyasr) [inline .pre-commit-config.yaml:55]:
```suggestion
    rev: v1.4.0
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -51,3 +51,7 @@ repos:
...
+    hooks:
+        - id: rapids-dependency-file-generator
```
Comment (vyasr) [inline .pre-commit-config.yaml:57]:
```suggestion
        - id: rapids-dependency-file-generator
          args: ["--clean"]
```

**Review by ajschmidt8 — APPROVED** [review]

**Pull request merged**